### PR TITLE
BytecodeCreator - add ifValue() to handle single-valued if statements

### LINKED ISSUE #39: Add more if statements
Currently, we only have `ifNonZero` and `ifNull`. It would be useful to add more variants, `ifgt` or `iflt` at least.

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #39

### CODE REVIEW

**Comment by mkouba** [comment]
It could be useful to add support for `if_icmpeq` and friends as well...

File: src/main/java/io/quarkus/gizmo/BytecodeCreator.java [hunk 1]
```diff
@@ -559,4 +571,62 @@ default void breakScope() {
      */
     @Override
     default void close() {}
+    
+    /**
+     * The JVM opcode that can be used for {@link BytecodeCreator#ifValue(ResultHandle, IfOpcode)}.
+     * 
+     * @see Opcodes
+     */
+    enum IfOpcode {
```
Comment (stuartwdouglas) [inline src/main/java/io/quarkus/gizmo/BytecodeCreator.java:580]: I don't really like this, you should not need to know about bytecodes to use Gizmo, I think we should add methods for each of them.

Comment (mkouba) [reply src/main/java/io/quarkus/gizmo/BytecodeCreator.java:580]: Well, you need to know something about bytecode (see for example all those `invokeX()` methods, `ifNonZero()` is used to test booleans, etc.). In fact, you need to know a lot about bytecode in order to produce "effective" classes. I can rework the PR though and add those 6 methods.

**Comment by mkouba** [comment]
I've added few more if statements...

**Pull request merged**